### PR TITLE
Add snapshot and release workflows

### PR DESCRIPTION
**Pull request by uglide**
Add new workflows to upload snapshots and release artifacts automatically.

### CODE REVIEW

File: .github/workflows/snapshot.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up publishing to maven central
+        uses: actions/setup-java@v4
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: Cache dependencies
```
Comment (tishun) [inline .github/workflows/snapshot.yaml:27]: Not that important:
We can skip this entire step if we use the built-in caching of the setup-java step (the previous one)
See cicd.yaml

File: .github/workflows/snapshot.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            ~/.m2/repository
+            /var/cache/apt
+          key: lettuce-${{hashFiles('**/pom.xml')}}
+      - name: mvn offline
+        run: |
+          mvn -q dependency:go-offline
+      - name: deploy
+        run: |
+          mvn --no-transfer-progress \
+            -DskipTests deploy
```
Comment (tishun) [inline .github/workflows/snapshot.yaml:40]: Out of curiosity: What is the rationale to skip the tests?

File: .github/workflows/snapshot.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      - '[0-9].x'
+  workflow_dispatch:
+
+jobs:
+
+  snapshot:
+    name: Deploy Snapshot
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (tishun) [inline .github/workflows/snapshot.yaml:18]: Let's use v4?

File: .github/workflows/version-and-release.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+on:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (tishun) [inline .github/workflows/version-and-release.yaml:12]: Let's use v4?

**Review by tishun — APPROVED** [review]

File: .github/workflows/snapshot.yaml (hunk 1 above)
Comment (uglide) [reply .github/workflows/snapshot.yaml]: Thanks for the suggestion! Done.

File: .github/workflows/snapshot.yaml (hunk 3 above)
Comment (uglide) [reply .github/workflows/snapshot.yaml]: Done

File: .github/workflows/version-and-release.yaml (hunk 4 above)
Comment (uglide) [reply .github/workflows/version-and-release.yaml]: Done

File: .github/workflows/snapshot.yaml (hunk 2 above)
Comment (uglide) [reply .github/workflows/snapshot.yaml]: Good question. Removed :)

File: .github/workflows/snapshot.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+          cache: 'maven'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: mvn offline
+        run: |
+          mvn -q dependency:go-offline
+      - name: deploy
+        run: |
+          mvn --no-transfer-progress deploy
```
Comment (mp911de) [inline .github/workflows/snapshot.yaml:33]: Adding `-U -B` ensures latest dependency versions and uses batch-mode to skip interactive elements.

**Comment by mp911de** [comment]
This is a neat approach. Left a minor comment.

**Review by sazzad16 — APPROVED** [review]

File: .github/workflows/snapshot.yaml (hunk 5 above)
Comment (uglide) [reply .github/workflows/snapshot.yaml:33]: On minor version branches, it might not be the behavior we want.

**Comment by tishun** [comment]
Closes #2752 
Closes #2753

**Pull request merged**